manifests/params.pp

```diff
@@ -10,7 +10,7 @@
   $package_ensure    = 'latest'
   $ui_package_name   = 'consul_ui'
   $ui_package_ensure = 'latest'
-  $version = '0.5.0'
+  $version = hiera('consul::version', '0.5.0')  # ensure defaults resolve from hiera as well
```

Comment:
It is not advised to call Hiera from a Module ...

Best practices are to use the Role / Profile pattern to compose nodes definition and make Hiera calls from these classes which are business specific anyway .